### ISSUE TITLE
Remove worker access checks

### ISSUE THREAD

**Issue by ThorbenLindhauer** [zeebe-io/zeebe#903]
We decided that a job's `worker` attribute has purely non-functional purposes.

The broker should not check that the worker in the command matches the currently assigned worker.

**Comment by Zelldon** [comment]
I thought that only the assigned worker can complete the job?

**Comment by ThorbenLindhauer** [comment]
While this is the current behavior, we decided in a recent team meeting that a job's `worker` attribute should should have a purely informational purpose. We did not see a reason why e.g. job completion should be restricted by worker id. If you believe that there is such a reason, then let's discuss this via this issue/slack/team meeting.

**Comment by Zelldon** [comment]
Yeah would like to discuss this, because for me it is more an safety guarantee. I can only complete jobs where I'm assigned to and only me can complete them. If we drop this then I see no benefit of assignment at all.

**Comment by Zelldon** [comment]
OK I thought about this will still kind of work since only one get this job via the subscription (however it is now called) right? The only problem is that if  complete uses the wrong job id, which does not belong to me.

**Comment by ThorbenLindhauer** [comment]
> I can only complete jobs where I'm assigned to and only me can complete them.

Not sure if this will ever practically benefit any user. Also, with https://github.com/zeebe-io/zeebe/issues/795 the complete command would be rejected anyway in most cases.

> If we drop this then I see no benefit of assignment at all.

Auditing. You can build an application and see which worker processed how many jobs when, etc.

**Comment by Zelldon** [comment]
> Auditing. You can build an application and see which worker processed how many jobs when, etc.
True.

> Not sure if this will ever practically benefit any user. Also, with #795 the complete command would be rejected anyway in most cases.

Hm yeah if it is implemented :D

**Comment by Zelldon** [comment]
Ok this would be possible with a topic subscription but yeah agree.
